### PR TITLE
feat: first version

### PR DESCRIPTION
**Pull request by Isammoc**
Motivation:
Customer asked for a base image for our dedicated load generator.

Modifications:
From a debian base image, install openssh-server, java zulu 21, curl, jq Add some configuration to let the final user chose the available public key and the final username.
Document how to launch the image
Create GitHub workflows to publish this new image

Result:
A new image will be available on docker hub.
This should ease customer that work with docker in their dedicated machines.

### CODE REVIEW

**Review by slandelle — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+FROM debian:bullseye
```
Comment (bastien-gatling) [inline Dockerfile]: I don't know if it's intentional, but the last debian version is bookworm.

File: root/docker/docker-entrypoint [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+# create folders
+mkdir -p \
+    /config/{.ssh,ssh_host_keys,logs/openssh} \
+    /run/sshd
+
+# symlink out ssh config directory
+if [[ ! -L /etc/ssh ]]; then
+    if [[ ! -f /config/ssh_host_keys/sshd_config ]]; then
+        sed -i '/#PidFile/c\PidFile \/config\/sshd.pid' /etc/ssh/sshd_config
```
Comment (bastien-gatling) [inline root/docker/docker-entrypoint:34]: Why ? We use this pid file ?

File: root/docker/docker-entrypoint [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        cp -a /etc/ssh/sshd_config /config/ssh_host_keys/
+    fi
+    rm -Rf /etc/ssh
+    ln -s /config/ssh_host_keys /etc/ssh
+    ssh-keygen -A
+fi
+
+# set umask for sftp
+UMASK=${UMASK:-022}
+sed -i "s|/usr/lib/ssh/sftp-server$|/usr/lib/ssh/sftp-server -u ${UMASK}|g" /etc/ssh/sshd_config
```
Comment (bastien-gatling) [inline root/docker/docker-entrypoint:44]: What's the default configuration? Where come from the need to modify it ?

**Review by bastien-gatling — APPROVED** [review]
Quite a few default configuration changes that I'm not sure are necessary.

Seem to do the job :).

File: Dockerfile (hunk 1 above)
Comment (Isammoc) [reply Dockerfile]: indeed... released on February 10th, 2024.
wasn't aware

File: root/docker/docker-entrypoint (hunk 3 above)
Comment (guilgaly) [reply root/docker/docker-entrypoint:44]: As far as I understand:
- OpenSSH's default is to simply use the user's default umask
- `-u` forces a specific umask instead, for all new files/directories created, regardless of the user doing it

File: root/docker/docker-entrypoint (hunk 2 above)
Comment (Isammoc) [reply root/docker/docker-entrypoint:34]: sshd itself

Comment (guilgaly) [reply root/docker/docker-entrypoint:34]: this part might need more detailed comments?

File: root/docker/docker-entrypoint (hunk 3 above)
Comment (Isammoc) [reply root/docker/docker-entrypoint:44]: ensure that the connected user can create AND modify the file they upload

**Review by guilgaly — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (Isammoc) [reply Dockerfile]: Tested and still work, updated.

Thanks :+1:

File: root/docker/docker-entrypoint (hunk 2 above)
Comment (Isammoc) [reply root/docker/docker-entrypoint:34]: To be honest, I got most of the code from [linuxserver](https://github.com/linuxserver/docker-openssh-server/)

File: README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+|                `-e PUBLIC_KEY_FILE=/path/to/file`                | Optionally specify a file in the container containing the public key (works with docker secrets).       |
+| `-e PUBLIC_KEY_DIR=/path/to/directory/containing/_only_/pubkeys` | Optionally specify a directory in the container containing the public keys (works with docker secrets). |
+|       `-e PUBLIC_KEY_URL=https://github.com/username.keys`       | Optionally specify a URL containing the public key.                                                     |
+|                   `-e USER_NAME=gatling-user`                    | Optionally specify a user name (Default:`gatling-user`)                                                 |
+
+You need at least to define one public key parameter.
+
+## What's inside
+
+ * debian bullseye base
```
Comment (abarreau) [inline README.md:36]: `debian bookworm base`

**Pull request merged**